### PR TITLE
001 view job roles

### PR DESCRIPTION
**Pull request by chrisknos**

### CODE REVIEW

**Review by 40175096 — CHANGES_REQUESTED** [review]
Lint check has failed

**Review by 40175096 — APPROVED** [review]

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+on:
+  push:
+  pull_request:
```
Comment (shaunganley) [inline .github/workflows/test.yml]: you're not specifying which branch you want to run against

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+jobs:
+  eslint:
```
Comment (shaunganley) [inline .github/workflows/test.yml]: is this for tests or linter? update these names if needed

File: test/ui/JobsRolesTests.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    Expect the page should display a valid list of job roles fetched from the /job-roles API endpoint.
+    */
+    it('Displays a list of job roles on the job roles page', async () => {
+
+        var driver = new webdriver.Builder().
+            withCapabilities(webdriver.Capabilities.chrome()).
+            build();
+
+        // Visit the job roles page
+        await driver.get("http://localhost:3000/jobs")
```
Comment (shaunganley) [inline test/ui/JobsRolesTests.js]: don't hardcode localhost, pass in as env var so we could run these against any URL

File: test/ui/JobsRolesTests.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    
+        // Add more specific assertions based on your UI structure
+        // For example, check if there are specific job roles listed
+        await driver.findElement(webdriver.By.xpath('//td[contains(text(), "Trainee Software Engineer")]'));
+    
+        // Check that the job roles are rendered in a certain way
+        // For example, each job role might be inside a table row with class 'job-role'
+        // Check that there is more than 0 table rows are returned
+        const tableRows = await driver.findElements(webdriver.By.css('tr.job-role'));
+        chai.assert.isAbove(tableRows.length, 0, 'Expected more than zero rows in the table');
```
Comment (shaunganley) [inline test/ui/JobsRolesTests.js]: you should only find things by ID, you may need to add IDs to the html in order to find them

**Review by shaunganley — CHANGES_REQUESTED** [review]

File: .github/workflows/test.yml (hunk 2 above)
Comment (eoinmcbennett) [reply .github/workflows/test.yml]: changed to be more specific to the test run

File: .github/workflows/test.yml (hunk 1 above)
Comment (eoinmcbennett) [reply .github/workflows/test.yml]: Changed to run generally on all push / pull request events

File: test/ui/JobsRolesTests.js (hunk 3 above)
Comment (chrisknos) [reply test/ui/JobsRolesTests.js]: Changed to using env variable

File: test/ui/JobsRolesTests.js (hunk 4 above)
Comment (chrisknos) [reply test/ui/JobsRolesTests.js]: Test and html updated in order to find job role by ID

File: .github/workflows/lint.yml [hunk 5]
```diff
@@ -1,6 +1,4 @@
-on:
-  push:
-  pull_request:
+on: [push,pull_request]
```
Comment (shaunganley) [inline .github/workflows/lint.yml]: these still don't specify the main branch, have a look at other examples

File: .github/workflows/test.yml [hunk 6]
```diff
@@ -0,0 +1,16 @@
+on: [push,pull_request]
```
Comment (shaunganley) [inline .github/workflows/test.yml]: these still don't specify the main branch, have a look at other examples

**Review by shaunganley — CHANGES_REQUESTED** [review]
looks like accessibility check is failing

File: .github/workflows/lint.yml (hunk 5 above)
Comment (chrisknos) [reply .github/workflows/lint.yml]: Lint.yml has been updated to specify the main branch

File: .github/workflows/test.yml (hunk 6 above)
Comment (chrisknos) [reply .github/workflows/test.yml]: Test.yml has been updated to specify the main branch

**Review by shaunganley — APPROVED** [review]

**Pull request merged**